examples/js/mobilenet.js

```diff
@@ -1,89 +1,89 @@
-let fs = require('fs');
-let Redis = require('ioredis');
-let Jimp = require('jimp');
+const fs = require('fs');
+const Redis = require('ioredis');
+const Jimp = require('jimp');
+const Helpers = require('./Helpers');
+
+// Simple configuration, all these files should exist already in the example folder
+const config = {
+   jsonClassIndex: 'imagenet_class_index.json',
+   modelFile: '../models/mobilenet_v2_1.4_224_frozen.pb',
+   inputPath: 'input',
+   outputPath: 'MobilenetV2/Predictions/Reshape_1',
+   imageHeight: 224,
+   imageWidth: 224,
+};
+
+const answerSet = [];
+const helpers  = new Helpers();
+const labels   = JSON.parse(fs.readFileSync(config.jsonClassIndex));
+const redis    = new Redis({ parser: 'javascript' });
+
+let progress = 0;
+
+function run(filenames) {
+
+   /**
+    * We're looping over each image and seeing if we can find a match
+    */
+   filenames.forEach((filename, key, filenames) => {
+      Jimp.read(filename).then((inputImage) => {
+         console.log(`\nWorking on ${filename}`);
+         let image = inputImage.cover(config.imageWidth, config.imageHeight);
+         let normalized = helpers.normalizeRGB(image.bitmap.data, image.hasAlpha());
+         let buffer = Buffer.from(normalized.buffer);
+
+         console.log("...Setting input tensor \n...running model");
+
+         /**
+          * Redis Pipelines, this allows us to set up and batch our commands
+          * If you want to learn more about how IORedis does this check out
+          * https://github.com/luin/ioredis#pipelining
+          */
+         redis.pipeline()
+              .call('AI.TENSORSET', 'input_' + key, 'FLOAT', 1, config.imageWidth, config.imageHeight, 3, 'BLOB', buffer)
+              .call('AI.MODELRUN', 'mobilenet', 'INPUTS', 'input_' + key, 'OUTPUTS', 'output_' + key)
```

Comment:
Here you're first setting the image to a key, then setting the output tensor to the same key. That's fine but maybe it's a bit counterintuitive, or the rationale could be explained in a comment.

I'm not saying it's wrong, it just needs to be explained I guess.

My first instinct for evaluating the model on a list of images and just get the result is to use the same input key and output key for all images in a sequence - if I'm just interested in the output there's no point in keeping them in different keys.

This works unless you want to keep results around, in which case you have to use different keys. At that point you might want to set and EXPIRE to those keys or take care of calling DEL on them when you don't need them anymore.

A comment on why you're using the same key for input and output and different keys for a sequence of images would be good.

Reply:
Yeah. My assumption here was that this data didn't need to be kept around, but this might give someone the impression that the keys should be the same or that they're tied together in some way in which they're not, I'll adjust

Reply:
...I have no idea how I dropped that!